### PR TITLE
Update to JMS Serializer 0.12

### PR DESCRIPTION
**Pull request by baldurrensch**
Required some changes because of the BC break with the `context`

### CODE REVIEW

File: Serializer/Handler/FormViewHandler.php [hunk 1]
```diff
@@ -54,8 +60,11 @@ public function serializeToXML(XmlSerializationVisitor $visitor, FormView $formV
             $visitor->document = $visitor->createDocument();
         }
 
-        $this->embedderEventSubscriber->onPostSerializeXML(new Event($visitor, $formView, $type));
-        $this->linkEventSubscriber->onPostSerializeXML(new Event($visitor, $formView, $type));
+        $context = SerializationContext::create();
```
Comment (stof) [inline Serializer/Handler/FormViewHandler.php]: you should not create a new context but use the current context. It does not make sense to use a different config here (forcing to use the default one). the context is passed as 4th argument to the method

Comment (baldurrensch) [inline Serializer/Handler/FormViewHandler.php]: I totally did not see that. Makes perfect sense. I changed all three occurrences.

File: composer.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
         "php": ">=5.3.0",
         "symfony/framework-bundle": ">=2.1,<2.3-dev",
         "jms/metadata": ">=1.1.0,<1.3-dev",
-        "jms/serializer": "0.11.*",
+        "jms/serializer": "0.12.*",
```
Comment (adrienbrault) [inline composer.json:15]: You should change the bundle branch alias to 0.4

File: Serializer/Handler/PagerfantaHandler.php [hunk 3]
```diff
@@ -7,6 +7,8 @@
 use JMS\Serializer\GraphNavigator;
 use JMS\Serializer\XmlSerializationVisitor;
 use JMS\Serializer\GenericSerializationVisitor;
+use JMS\Serializer\Context;
+use JMS\Serializer\EventDispatcher\ObjectEvent;
```
Comment (adrienbrault) [inline Serializer/Handler/PagerfantaHandler.php:10]: You can remove `use JMS\Serializer\EventDispatcher\Event;` right ?

File: Serializer/Handler/FormViewHandler.php [hunk 4]
```diff
@@ -32,16 +35,19 @@ public static function getSubscribingMethods()
...
     public function __construct(XmlFormViewSerializer $xmlFormViewSerializer,
-        EmbedderEventSubscriber $embedderEventSubscriber, LinkEventSubscriber $linkEventSubscriber)
-    {
+        EmbedderEventSubscriber $embedderEventSubscriber, LinkEventSubscriber $linkEventSubscriber,
+        MetadataFactoryInterface $serializerMetadataFactory
+    ) {
         $this->xmlFormViewSerializer = $xmlFormViewSerializer;
         $this->embedderEventSubscriber = $embedderEventSubscriber;
         $this->linkEventSubscriber = $linkEventSubscriber;
+        $this->serializerMetadataFactory = $serializerMetadataFactory;
```
Comment (adrienbrault) [inline Serializer/Handler/FormViewHandler.php]: Where is this used ?

File: Serializer/Handler/HalPagerfantaHandler.php [hunk 5]
```diff
@@ -29,29 +33,32 @@ public static function getSubscribingMethods()
...
     protected $relationsJsonKey;
 
     public function __construct(
+        MetadataFactoryInterface $serializerMetadataFactory,
         EmbedderEventSubscriber $embedderEventSubscriber,
         LinkEventSubscriber $linkEventSubscriber,
         $linksKey = null,
         $relationsKey = null
     ) {
+        $this->serializerMetadataFactory = $serializerMetadataFactory;
```
Comment (adrienbrault) [inline Serializer/Handler/HalPagerfantaHandler.php]: And where is this one used ?

File: composer.json (hunk 2 above)
Comment (baldurrensch) [inline composer.json:15]: Done

File: Serializer/Handler/PagerfantaHandler.php (hunk 3 above)
Comment (baldurrensch) [inline Serializer/Handler/PagerfantaHandler.php:10]: Yes, I removed them in the other places as well

File: Serializer/Handler/FormViewHandler.php (hunk 4 above)
Comment (baldurrensch) [inline Serializer/Handler/FormViewHandler.php]: Left over from when I was creating the serialization context myself. Removed.

File: Serializer/Handler/HalPagerfantaHandler.php (hunk 5 above)
Comment (baldurrensch) [inline Serializer/Handler/HalPagerfantaHandler.php]: Same

**Comment by Ninir** [comment]
@adrienbrault Any ~~change~~ **chance** to get this be merged by the end of the day/week please?

**Comment by adrienbrault** [comment]
Thanks!

**Pull request merged**